tendrl/monitoring_integration/graphite/__init__.py

```diff
@@ -192,127 +179,80 @@ def get_central_store_data(self, objects):
                        {'message': str(ex)})
             raise ex
 
-    def get_cluster_details(self, objects, cluster_key):
+    def get_cluster_details(self, objects, integration_id):
         cluster_detail = []
         for obj in objects["Cluster"]:
             if obj in ["metric", "value"]:
                 continue
             resource_detail = {}
             resource_detail[str(obj)] = {}
             obj_details = objects["Cluster"][str(obj)]
-            obj_key = os.path.join(cluster_key, str(obj))
             obj_attrs = obj_details["attrs"]
+            _resource_obj = NS.tendrl.objects[str(obj)](
+                integration_id=integration_id
+            ).load()
             for key, _ in obj_attrs.items():
-                try:
-                    attr_key = os.path.join(obj_key, key)
-                    attr_data = etcd_utils.read(attr_key)
-                    attr_value = self.cluster_status_mapper(
-                        str(attr_data.value))
-                    resource_detail[str(obj)][key] = copy.deepcopy(
-                        attr_value
-                    )
-                except (KeyError, etcd.EtcdKeyNotFound) as ex:
-                    integration_id = cluster_key.split("/")[-1]
-                    logger.log(
-                        "debug",
-                        NS.get("publisher_id", None),
-                        {'message': "Cannot Find {0} in Cluster "
-                         "{1}".format(key, integration_id) + str(ex)
-                         }
-                    )
+                attr_value = self.cluster_status_mapper(
+                    str(getattr(_resource_obj, key))
+                )
+                resource_detail[str(obj)][key] = copy.deepcopy(
+                    attr_value
+                )
             if not resource_detail == {}:
                 cluster_detail.append(resource_detail)
         return cluster_detail
 
-    def get_brick_details(self, objects, cluster_key):
+    def get_brick_details(self, objects, integration_id):
         brick_detail = []
-        host_list = utils.get_resource_keys(cluster_key, "Bricks/all")
-        for host in host_list:
-            attr_key = os.path.join(cluster_key, "Bricks/all", host)
-            brick_list = utils.get_resource_keys("", attr_key)
-            for brick in brick_list:
+        _cluster_node_ids = etcd_utils.read(
+            "/clusters/%s/nodes" % integration_id
+        )
+        for _node_id in _cluster_node_ids.leaves:
```

Comment:
@shtripat Why are you reading node_context and then brick dir you can directly do
etcd_utils.read("/cluster/cid/Bricks/all")  and you can find the node and then use can use GlusterBrick object.

Reply:
Loading all the bricks for a cluster is not possible because the path if like `/clusters/{int-id}/Bricks/all/{host-name}/{brick-path}`. So the load_all function removes the last entry from `/` and expects rest path as collection of bricks. That's not correct in this case and so `load_all()` returns None.
Explicitly I had to do this for that reason only.